src/arrayMethodSort.js

```diff
@@ -4,8 +4,34 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    if (typeof compareFunction === 'function') {
+      for (let i = 0; i < this.length; i++) {
+        for (let j = i + 1; j < this.length; j++) {
+          if (compareFunction(this[i], this[j]) > 0) {
+            const temp = this[j];
+
+            this[j] = this[i];
+            this[i] = temp;
+          }
+        }
+      }
```

Comment:
The current implementation uses a bubble sort algorithm, which is not efficient for large arrays. Consider implementing a more efficient sorting algorithm like quicksort or mergesort for better performance.